google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/PubSubImpl.java

```diff
@@ -512,110 +503,17 @@ public SubscriptionId apply(String compositeSubscription) {
     return listSubscriptionsAsync(topic, getOptions(), optionMap(options));
   }
 
-  private Future<Iterator<ReceivedMessage>> pullAsync(final String subscription,
-      int maxMessages, boolean returnImmediately) {
-    PullRequest request = PullRequest.newBuilder()
-        .setSubscription(
-            SubscriberClient.formatSubscriptionName(getOptions().getProjectId(), subscription))
-        .setMaxMessages(maxMessages)
-        .setReturnImmediately(returnImmediately)
-        .build();
-    PullFuture future = rpc.pull(request);
-    future.addCallback(new PubSubRpc.PullCallback() {
-      @Override
-      public void success(PullResponse response) {
-        List<String> ackIds = Lists.transform(response.getReceivedMessagesList(),
-            MESSAGE_TO_ACK_ID_FUNCTION);
-        ackDeadlineRenewer.add(subscription, ackIds);
-      }
-
-      @Override
-      public void failure(Throwable error) {
-        // ignore
-      }
-    });
-    return transform(future, new Function<PullResponse, Iterator<ReceivedMessage>>() {
-      @Override
-      public Iterator<ReceivedMessage> apply(PullResponse response) {
-        return Iterators.transform(response.getReceivedMessagesList().iterator(),
-            new Function<com.google.pubsub.v1.ReceivedMessage, ReceivedMessage>() {
-              @Override
-              public ReceivedMessage apply(com.google.pubsub.v1.ReceivedMessage receivedMessage) {
-                // Remove consumed message from automatic ack deadline renewer
-                ackDeadlineRenewer.remove(subscription, receivedMessage.getAckId());
-                return ReceivedMessage.fromPb(PubSubImpl.this, subscription, receivedMessage);
-              }
-            });
-      }
-    });
-  }
-
-  @Override
-  public Iterator<ReceivedMessage> pull(String subscription, int maxMessages) {
-    return get(pullAsync(subscription, maxMessages, true));
-  }
-
   @Override
-  public Future<Iterator<ReceivedMessage>> pullAsync(String subscription, int maxMessages) {
-    return pullAsync(subscription, maxMessages, false);
-  }
-
-  @Override
-  public MessageConsumer pullAsync(String subscription, MessageProcessor callback,
-      PullOption... options) {
-    Map<Option.OptionType, ?> optionMap = optionMap(options);
-    return MessageConsumerImpl.builder(getOptions(), subscription, ackDeadlineRenewer, callback)
-        .maxQueuedCallbacks(MAX_QUEUED_CALLBACKS.getInteger(optionMap))
-        .executorFactory(EXECUTOR_FACTORY.getExecutorFactory(optionMap))
+  public Subscriber subscriber(SubscriptionInfo subscription, Subscriber.MessageReceiver receiver) {
+    // TODO(pongad): Provide a way to pass in the rest of the options.
```

Comment:
@pongad , we probably want the input of @davidtorres on this matter: how many executors and channels to we want, in relation to the number of objects in the client library?